contracts/Vault.sol

```diff
@@ -76,43 +77,49 @@ contract Vault is ExchangeRate, Initializable, OwnableUpgradeSafe {
     /// @notice Deposits `_amount` of `token` into Aave pool and mints equivalent bTokens. 
     /// @param _amount The amount to deposit.
     function deposit(uint256 _amount) public {
+        require(_amount > 0, "amount to deposit cannot be zero");
         _stateTransition();
 
         token.transferFrom(msg.sender, address(this), _amount);
         token.approve(address(aavePool), _amount);
+        uint256 beforeBalance = aToken.balanceOf(address(this));
         aavePool.deposit(
             address(token),
             _amount,
             address(this),
             0
         );
+        require(aToken.balanceOf(address(this)) >= beforeBalance, "atoken balance after deposit must be more than before");
+        uint256 atokenAmount = aToken.balanceOf(address(this)).sub(beforeBalance);
```

Comment:
smart contract newbie question here. we call aToken.balanceOf() on lines 92 and 93. i believe this makes our contract more expensive to run. what is you just fetch the balance 1 time, store it in a function local variable and use it in lines 92 and 93?

Reply:
> i can see division causing precision problems, but not subtraction. what scenario would cause this? a whole number minus a whole number is always a whole number.

excellent point, and you are correct on this one.

I have revisited the problem, it was not in fact Kovan specific but in fact has to do with the size of the deposits. If I deposit in wei values, ie 1000 wei amount of DAI, then this revert message gets emitted. On the other hand if I was to deposit 1000 in eth amount (1000e18) this issue goes away. My hunch is that this has to do with interest calculation somehow. I will make a GitHub issue for this since it's a sizeable problem.

Reply:
> this makes our contract more expensive to run. what is you just fetch the balance 1 time, store it in a function local variable and use it in lines 92 and 93?

👍